README.md
<a href="https://www.zahradnik.io/" target="_blank">
<img src="https://raw.githubusercontent.com/zahradnik-io/gatsby-theme-novela-example/main/assets/gatsby-theme-novela-hero.jpg" alt="gatsby-novela-theme hero image" />
</a>

<br/>

# Novela — A Gatsby theme by Narative (fork by vzahradnik)
[![Build](https://github.com/zahradnik-io/gatsby-theme-novela/actions/workflows/build.yml/badge.svg?branch=main)](https://github.com/zahradnik-io/gatsby-theme-novela/actions/workflows/build.yml)

With minimal styling and maximum features — including multiple homepage layouts, built-in social sharing and dark mode — Novela makes it easy to start publishing beautiful articles and stories with Gatsby.

Novela was built by the team at [Narative](https://www.narative.co) for everyone that loves the web. I, Vladimír, will continue maintaining and improving the theme for the needs of my personal blog site. Feel free to contribute. Let's make this project thrive again.

Special thanks to [@jpvalery](https://github.com/jpvalery) for helping maintain Novela.

<div>
<a href="https://www.zahradnik.io/" target="_blank">
<img src="https://raw.githubusercontent.com/zahradnik-io/gatsby-theme-novela-example/main/assets/gatsby-theme-novela-cta-demo.jpg" alt="gatsby-novela-theme live demo" width="295px" />
</a>
</div>

<div>
<a href="https://www.narative.co/design/open/novela" target="_blank">
<img src="https://raw.githubusercontent.com/zahradnik-io/gatsby-theme-novela-example/main/assets/gatsby-theme-novela-cta-figma.jpg" alt="gatsby-novela-theme figma link" width="295px" />
</a>
</div>

### Table of Contents

- [Why Novela?](#why-use-novela)

- [Getting Started](#getting-started-with-gatsby-starter-novela)

  - With Gatsby Starter Novela

    - [Installation](#step-1-starter-installation)
    - [Develop & Build](#step-2-develop--build)

  - From scratch

    - [Installation](#step-1-installation)
    - [Folder structure](#step-2-folder-structure)
    - [Using Novela Theme](#step-3-using-narativegatsby-theme-novela)
    - [Adding an Author](#step-4-adding-an-author)
    - [Adding a Post](#step-5-adding-a-post)
    - [Configuring Site Metadata](#step-6-configuring-sitemetadata)

- [Data Sources](#data-sources)

  - [Local](#local)
  - [Contentful](#contentful)

- [Customization](#customization)

  - [Enabling Author Pages](#enabling-author-pages)
  - [Changing styles](#changing-styles)
  - [Component shadowing](#component-shadowing)
  - [Using images](#using-images)
  - [Adding your logo](#adding-your-logo)
  - [Adding Mailchimp](#adding-mailchimp)

- [Data Models](#data-models)

  - [Theme Options](#theme-options)
  - [Author](#author)
  - [Post](#post)
  - [Site Metadata](#site-metadata)

- [Future](#the-future)

  <br />

# Getting Started with Gatsby Starter Novela

This guide will take you through setting up Novela with Gatsby Starter Novela.

### Step 1: Starter installation

##### With `gatsby-cli`:

```sh
gatsby new novela-site https://github.com/zahradnik-io/gatsby-starter-novela
```

##### With `git clone`:

```sh
git clone git@github.com:zahradnik-io/gatsby-starter-novela.git novela-site

cd novela-site

yarn
```

### Step 2: Develop & Build

Once installed or cloned locally and all packages are installed you can begin developing your site.

```sh
# Run localhost
yarn dev

# Build your Gatsby site
yarn build
```

To learn more about adding Authors, Posts, and Site Metadata see:

- [Adding an Author](#step-4-adding-an-author)
- [Adding a Post](#step-5-adding-a-post)
- [Configuring Site Metadata](#step-6-configuring-sitemetadata)

# Getting Started from scratch

This guide will take you through adding Novela to a new project. You do not require any Gatsby starters or similar, but you can add Novela to an existing Gatsby project.

You can also view the completed [example repository](https://github.com/zahradnik-io/gatsby-theme-novela-example).

### Step 1: Installation

```sh
# Create an empty directory and go into it
mkdir novela-site && cd novela-site

# Add all required dependencies
yarn add react react-dom gatsby @zahradnik-io/gatsby-theme-novela
```

### Step 2: Folder structure

Once you've installed React, Gatsby, and Novela you'll want to add your first Author and Post.

The recommended project structure for your content and site looks like this:

```
  novela-site
    ├── content
    │ ├── authors
    │ │   ├── avatars
    │ │   │    └── avatar.jpg
    │ │   └── authors.yml
    │ └── posts
    │     └── 2020-01-01-my-first-novela-post
    │         ├── images
    │         │   └── novela-hero.jpg
    │         └── index.mdx
    ├── node_modules
    ├── gatsby-config.js
    └── package.json
```

### Step 3: Using `@zahradnik-io/gatsby-theme-novela`

You must add `@zahradnik-io/gatsby-theme-novela` as a plugin in `gatsby-config.js`.
`@zahradnik-io/gatsby-theme-novela` is configurable through plugin options but we will use the defaults for now.

```js
// gatsby-config.js

module.exports = {
  plugins: ['@zahradnik-io/gatsby-theme-novela'],
};
```

---

Once you've setup `@zahradnik-io/gatsby-theme-novela` plugin in `gatsby-config.js` you can start creating your first Posts. In order to create a Post you also need at least one Author.

### Step 4: Adding an Author

In [step 2](#step-2-folder-structure) we created the folder structure of our project. We can now add an Author by populating `/content/authors/authors.yml`:

```
  novela-site
  └── content
    └── authors
        ├── avatars
        │    └── brotzky-avatar.jpg
        └── authors.yml
```

In `authors.yml` add an Author. There **must** be at least one `featured` Author.

`/content/authors/authors.yml`:

```yml
- name: Dennis Brotzky
  bio: |
    Written by Dennis Brotzky who lives and works in Vancouver building useful things.
    You should follow him on Twitter.
  avatar: ./avatars/brotzky-avatar.jpg
  featured: true
  social:
    - url: https://unsplash.com
    - url: https://stackoverflow.com
    - url: https://github.com
```

### Step 5: Adding a Post

Once you have at least one Author defined in `authors.yml` you can add your first Post.

Start by creating a new folder in `content/posts`. You can name it anything you like but we recommend including the date at the front to organize your posts. Once you've created your folder you can add an `index.mdx` file and an `images` folder.

`/content/posts/2020-01-01/index.mdx`

```yml
---
title: Why Narative loves Gatsby
author: Dennis Brotzky
date: 2019-04-27
hero: ./images/narative-gatsby-hero.jpg
excerpt: This is a love story about Narative and Gatsby
---
# And then under the heading YML you can insert any MDX you like
# like headings, links, code, images, etc
# This will show up in the body of your post
# ...
```

### Step 6: Configuring siteMetadata

In order to configure the theme to properly generate the pages and meta tags you must add specific data to `siteMetadata`.

The fields that are unique to Novela are `hero.heading`, `hero.maxWidth`, and `social`.

Add your Site Metadata to the `gatsby-config.js` file.

```js
// gatsby-config.js

module.exports = {
  siteMetadata: {
    title: `Novela by zahradnik-io`,
    name: `Vladimír Záhradník`,
    siteUrl: `https://gatsby-theme-novela.netlify.com`,
    description: `This is my description that will be used in the meta tags and important for search results`,

    // important to set the main text that appears in the hero
    hero: {
      heading: `Perspectives on technology, design and business from Vladimír Záhradník.`,
      maxWidth: 652,
    },
    social: [
      {
        name: `twitter`,
        url: `https://twitter.com/VladoZahradnik`,
      },
      {
        name: `github`,
        url: `https://github.com/zahradnik-io`,
      },
    ],
  },
  plugins: ['@zahradnik-io/gatsby-theme-novela'],
};
```

### You can now run your site

Once all steps have been completed you can run your site. In the root of your project run `gatsby develop`.

If you ran into problems you can reference the [example repository](https://github.com/zahradnik-io/gatsby-theme-novela-example) or create an issue.

<br />

- [Data Sources](#data-sources)

# Data Sources

With the flexibility of Gatsby, Novela is able to pull different data sources to build your website. Right now we support Contentful and your local file system. All data sources can be combined together or used separately.

### Local

Local is the default data source for Novela. Write MDX and YAML in order to generate posts and authors. This is the fastest and simplest way to get started.

To learn how to use the local file system, read the [Installation](#installation) steps.

### Contentful

Contentful provides the flexibility of a headless CMS which allows you to write content without committing new files and working in a text editor. The setup process is simple if you are familiar with Contentful.

You will need to setup your Contentful space and import the Novela model.

#### Setting up `gatsby-source-contentful` in your project

In order to use Contentful you must first install the plugin and `dotenv` in your project

```sh
yarn add gatsby-source-contentful dotenv
```

Then pass in the environment variables `.env` to the plugin

#### `.env`

```
CONTENTFUL_SPACE_ID= (Your Contentful Space ID)
CONTENTFUL_ACCESS_TOKEN= (Your Content Delivery API - access token)
```

#### `gatsby-config.js`

```js
require('dotenv').config();

plugins: [
  {
    resolve: 'gatsby-source-contentful',
    options: {
      spaceId: process.env.CONTENTFUL_SPACE_ID,
      accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
    },
  },
  {
    resolve: '@zahradnik-io/gatsby-theme-novela',
    options: {
      sources: {
        contentful: true,
      },
    },
  },
];
```

Finally, import the Contentful Model from Novela to get started. Novela Contentful Queries will not work without this exact data model.

#### [Novela Contentful Data](https://github.com/zahradnik-io/gatsby-theme-novela/tree/main/%40zahradnik-io/gatsby-theme-novela/contentful)

[Contentful space import docs](https://github.com/contentful/contentful-cli/tree/master/docs/space/import)

Once you have your project setup with `gatsby-source-contentful` and `@zahradnik-io/gatsby-theme-novela` and your Contentful space is setup with the imported model you are good to go.

⚠️ Please make sure you have at least one Media Asset uploaded or you will run into a GraphQL error: https://github.com/gatsbyjs/gatsby/issues/15397

# Customization

Once you've created the Logo component it should automatically appear in your site.

The technique we have used is called Component Shadowing and is a core feature of Gatsby Themes.

### Enabling Author pages

By default Author pages are not enabled. They can be enabled through the plugin option `authorsPage`. If you decided to enable Author pages please make sure you have all the required for each author you add.

```js
module.exports = {
  plugins: [
    {
      resolve: '@zahradnik-io/gatsby-theme-novela',
      options: {
        authorsPage: true,
      },
    },
  ],
};
```

### Changing styles

Novela allows you to change the default theme styling by updating the [theme-ui](https://theme-ui.com/) values. If you're familiar with Styled Components or Emotion it's the same as adapting the theme you pass to `ThemeProvider`.

First, you must create a theme file and then you can override `novelaTheme` values.
[See all Novela theme values](https://github.com/zahradnik-io/gatsby-theme-novela/tree/main/%40zahradnik-io/gatsby-theme-novela/src/gatsby-plugin-theme-ui)

```js
// src/gatsby-plugin-theme-ui/index.js

import novelaTheme from '@zahradnik-io/gatsby-theme-novela/src/gatsby-plugin-theme-ui';

export default {
  ...novelaTheme,
  initialColorModeName: `dark`,
  colors: {
    ...novelaTheme.colors,
    primary: '#000',
    secondary: '#73737D',
    accent: '#6166DC',
    grey: '#73737D',
    background: '#fff',
  },
};
```

### Component Shadowing

> This feature allows users to override a component in order to customize its rendering.

> Component Shadowing lets you replace the theme’s original file, gatsby-theme-novela/src/components/Logo.js, with your own to implement any changes you need.

Any [component](https://github.com/zahradnik-io/gatsby-theme-novela/tree/main/%40zahradnik-io/gatsby-theme-novela/src/components) or [section](https://github.com/zahradnik-io/gatsby-theme-novela/tree/main/%40zahradnik-io/gatsby-theme-novela/src/sections) is able to be replaced with your own custom component.

This opens up a full customization of Novela to your designed needs. You can copy any component directly from Novela and alter it how you like, or you can create your own component to replace Novela's entirely.

A basic example of component shadowing is explained below by [adding your logo](#adding-your-logo).

### Using images

Images can be added to Posts and customized to fit your content. Define a regular Markdown image that will default to Small or write HTML to customize the sizes. All images can be clicked to zoom.

```html
![This is the alt text for small image](./images/small.jpg)

<div className="Image__Small">
  <img src="./images/small.jpg" alt="This is the alt text small image" />
</div>

<div className="Image__Medium">
  <img src="./images/medium.jpg" alt="This is the alt text medium image" />
</div>

<div className="Image__Large">
  <img src="./images/large.jpg" alt="This is the alt text large image" />
</div>
```

| Size   |   Class Name    |          Description           |
| ------ | :-------------- | :---------------------------- |
| small  | Image\_\_Small  |       Width of the text       |
| medium | Image\_\_Medium | Larger than width of the text |
| large  | Image\_\_Large  |       Full width image        |

### Adding your logo

Your logo must be in SVG (vector) format in order to add it to the theme. This is required because we will be making a React component containing your SVG Logo.

Start by creating the component file at:

```
  novela-site
  └── src
    └── @zahradnik-io
      └── gatsby-theme-novela
        └── components
          └── Logo
            └── index.js
```

It is important you create the exact folder structure so Gatsby knows to shadow this component. Once the file is created you can create your Logo component.

```jsx
import React from 'react';

/**
 * Paste in your SVG logo and return it from this component.
 * Make sure you have a height set for your logo.
 * It is recommended to keep the height within 25-35px.
 * Logo comes with a property value called `fill`. `fill` is useful 
 * when you want to change your logo depending on the theme you are on. 
 */
export default function Logo({ fill }) {
  return (
    <svg viewBox="0 0 106 28" height="30px">
      <path d="M62.9 12h2.8v10...." />
      <path fill={fill} d="M25 14h-..." />
      <path d="M14 0C6.3..." fill="#639" />
    </svg>
  );
}
```

### Adding Mailchimp

Want to add a subscription newsletter to your Posts? Novela gives you the option to integrate and build your following in a few easy steps.

Start by downloading [gatsby-plugin-mailchimp](https://github.com/benjaminhoffman/gatsby-plugin-mailchimp):

```sh
yarn add gatsby-plugin-mailchimp
```

Then configure the plugins

```js
plugins: [
  {
    resolve: '@zahradnik-io/gatsby-theme-novela',
    options: {
      contentPosts: 'content/posts',
      contentAuthors: 'content/authors',
      basePath: '/',
      mailchimp: true, // make sure this is true!
    },
  },
  {
    resolve: 'gatsby-plugin-mailchimp',
    options: {
      endpoint: '', // add your MC list endpoint here; see plugin repo for instructions
    },
  },
];
```

That's it. You will now have subscription boxes on each of your Posts.
To disable the subscription box on individual Posts you can set `subscription: false` on the Post.

You can shadow the Subscription component to customize the text displayed. It is recommended to copy and paste the current component and only alter the text.

You will want to override it here:

```
  novela-site
  └── src
    └── @zahradnik-io
      └── gatsby-theme-novela
        └── components
          └── Subscription
            └── index.js
```

<br />

# Data Models

## Theme Options

It is recommended to use the Default options, but if your project requires something else you can configure them to your need.

| Option                 |     Default     |                                                             Description                                                             |
| ---------------------- | :------------- | :--------------------------------------------------------------------------------------------------------------------------------- |
| contentPosts           |  content/posts  |                                          Define where you want to pull your Post data from                                          |
| contentAuthors         | content/authors |                                         Define where you want to pull your Author data from                                         |
| authorsPage            |      false      |                                                         Create Author pages                                                         |
| authorsPath            |    /authors     |                                                   Where should Author pages live?                                                   |
| rootPath               |        /        |                                                   Define the index of your site                                                     |
| basePath               |        /        |                      Where should the site be served from? `/blog` will change all paths to start with `/blog`                      |
| pageLength               |        6        |                      How many posts should be shown per page                      |
| articlePermalinkFormat |      :slug      | Define the format of the article permalink. Possible values: `:slug`, `:year`, `:month`, `:day`. Example: `:year/:month/:day/:slug` |
| mailchimp              |      false      |                                             Enable Mailchimp subscriptions on each Post                                             |
| sources.local          |      true       |                                                Enable local file system data source                                                 |
| sources.contentful     |      false      |                                                    Enable Contentful data source                                                    |

[View Theme option example](https://github.com/zahradnik-io/gatsby-theme-novela-example/blob/main/gatsby-config.js#L36)

```js
plugins: [
  {
    resolve: '@zahradnik-io/gatsby-theme-novela',
    options: {
      contentPosts: 'content/posts',
      contentAuthors: 'content/authors',
      rootPath: '/',
      basePath: '/',
      pageLength: 6,
      mailchimp: true,
      sources: {
        local: true,
        contentful: true,
      },
    },
  },
];
```

## Author

[View Author example](https://github.com/zahradnik-io/gatsby-theme-novela-example/blob/main/content/authors/authors.yml)

| Key      | Required |  Type   |                                                                 Description                                                                 |
| -------- | :------: | :----- | :---------------------------------------------------------------------------------------------------------------------------------------- |
| name     | required | String  |                                The Author's full name which is used should be used as a reference in Posts                                 |
| bio      | required | String  |                                            The Author's bio which is displayed on the home page                                            |
| avatar   | required |  Image  |                                                            The Author's avatar                                                             |
| featured | optional | Boolean |                                              If `true` the Author will appear on the homepage                                              |
| slug     | optional | String  |                          Override the autogenerated slug based on the Author's name. Do not include any slashes.                           |
| social   | required |  Array  | A list of social accounts and urls. [View supported icon/name combinations](https://github.com/zahradnik-io/gatsby-theme-novela/blob/main/%40zahradnik-io/gatsby-theme-novela/src/components/SocialLinks/SocialLinks.tsx#L15) |

```yml
- name: Dennis Brotzky
  bio: |
    Written by You. This is where your author bio lives. Share your work, your
    joys and of course, your Twitter handle.
  avatar: ./avatars/dennis-brotzky.jpg
  featured: true
  social:
    - url: https://github.com
    - url: https://twitter.com

- name: Thiago Costa
  bio: |
    Written by You. This is where your author bio lives. Share your work, your
    joys and of course, your Twitter handle.
  avatar: ./avatars/thiago-costa.png
```

## Post

| Key          | Required |    Type    |                                          Description                                          |
| ------------ | :------: | :-------- | :------------------------------------------------------------------------------------------- |
| title        | required |   String   |                  Used as title and generates a default slug. Must be unique.                  |
| slug         | optional |   String   |                Define a custom slug that will override the default title slug.                |
| author       | required | String Ref | Must **match** a defined Author name. Co-author posts by adding comma seperated Author names. |
| date         | required |    Date    |                                       YYYY-MM-DD format                                       |
| hero         | required |   Image    |                               1200px minimum width recommended                                |
| excerpt      | required |   String   |                                      140 character limit                                      |
| subscription | optional |   String   |          If mailchimp is enabled disable the subscription box on an individual Post           |
| secret       | optional |  Boolean   |           If secret the Post will not appear in paginated lists. Defaults to false.           |

[View Post example](https://github.com/zahradnik-io/gatsby-theme-novela-example/blob/main/content/posts/2019-04-31-understanding-the-gatsby-lifecycle/index.mdx)

```yml
# novela-site/content/posts/2020-01-01/index.mdx
---
title: Why Narative loves Gatsby
author: Dennis Brotzky, Thiago Costa, Brad Tiller
date: 2019-04-27
hero: ./images/narative-gatsby-hero.jpg
excerpt: This is a love story about Narative and Gatsby
---
# And then under the heading YML you can insert any MDX you like
# like headings, links, code, images, etc
# This will show up in the body of your post
# ...
```

## Site Metadata

| Key           | Required |    Type    |                                                                     Description                                                                     |
| ------------- | :------: | :-------- | :------------------------------------------------------------------------------------------------------------------------------------------------- |
| title         | required |   String   |                                                            Used for the <title></title>                                                             |
| name          | required | String Ref |                                           Used in multiple locations including meta tags and site footer                                            |
| siteUrl       | required |    Date    |                                                                  Used in meta tags                                                                  |
| description   | required |   String   |                                                                  Used in meta tags                                                                  |
| hero.heading  | required |   String   |                                                                  Used in the Hero                                                                   |
| hero.maxWidth | optional |   number   |                                                          Used in the Hero. Defaults to 652                                                          |
| social        | required |   Array    | [View supported icon/name combinations](https://github.com/zahradnik-io/gatsby-theme-novela/blob/main/%40zahradnik-io/gatsby-theme-novela/src/components/SocialLinks/SocialLinks.tsx#L15). `url` is required and the icon is chosen based on the url. If you also provide `name`, then the icon will be the same as the name you provided as long as the name equals one of the supported icons (See example bellow). Used in site footer and meta tags |

[View Site Metadata example](https://github.com/zahradnik-io/gatsby-theme-novela-example/blob/main/gatsby-config.js)

Within `gatsby-config.js` you can configure `siteMetadata` to show the values you wish.
It is `required` to add `siteMetada`

```js
// gatsby-config.js
module.exports = {
  siteMetadata: {
    title: `Novela by zahradnik-io`,
    name: `Vladimír Záhradník`,
    siteUrl: `https://gatsby-theme-novela.netlify.com`,
    description: `This is my description that will be used in the meta tags and important for search results`,

    // hero
    // A required key and will be displayed on the main page of Noveal
    hero: {
      heading: `Perspectives on technology, design and business from Vladimír Záhradník.`,
      maxWidth: 652,
    },

    // social
    // Add in the social links that will be displayed in the footer
    social: [
      {
        url: `https://twitter.com/VladoZahradnik`,
      },
      {
        url: `https://github.com/zahradnik-io`,
      },
      {
        url: `https://www.instagram.com/vladimir.zahradnik/`,
      },
      {
        name: 'stackoverflow',
        url: `https://stackoverflow.com/users/3296749/vzahradnik`,
      },
    ],
  },
};
```

## Why use Novela?

There are many Gatsby themes to choose from. Here’s why we think you won’t regret choosing Novela:

### Multiple Homepage Layouts

Choose between a variable width grid or a simpler list style to display each story.

### Mailchimp integration

Start building an audience and grow your blog subscribers with Mailchimp.

### Toggleable Light and Dark Mode

Out of the box, Novela includes both light and dark designs that can be toggled by the user anywhere across the site.

<img src="https://raw.githubusercontent.com/zahradnik-io/gatsby-theme-novela-example/main/assets/gatsby-theme-novela-light-dark.gif" alt="gatsby-novela-theme light dark theme demonstration" />

### Simple Customization with [Theme UI](https://theme-ui.com/)

Consistent, easy-to-read code lets you quickly customize every color and setting.

### Show code effortlessly

High quality embedded codeblocks that make authoring technical blog posts a breeze.

### Highlight-to-Share

Users can select text within an article to copy or share to platforms like Twitter and LinkedIn.


www/CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.6.14](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.13...www@1.6.14) (2021-03-24)

**Note:** Version bump only for package www





## [1.6.13](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.12...www@1.6.13) (2021-02-11)

**Note:** Version bump only for package www





## [1.6.12](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.11...www@1.6.12) (2020-12-29)

**Note:** Version bump only for package www





## [1.6.11](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.10...www@1.6.11) (2020-10-31)

**Note:** Version bump only for package www





## [1.6.10](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.9...www@1.6.10) (2020-10-27)

**Note:** Version bump only for package www





## [1.6.9](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.8...www@1.6.9) (2020-10-20)

**Note:** Version bump only for package www





## [1.6.8](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.7...www@1.6.8) (2020-10-20)

**Note:** Version bump only for package www





## [1.6.7](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.6...www@1.6.7) (2020-09-14)

**Note:** Version bump only for package www





## [1.6.6](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.5...www@1.6.6) (2020-07-30)

**Note:** Version bump only for package www





## [1.6.5](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.4...www@1.6.5) (2020-07-08)

**Note:** Version bump only for package www





## [1.6.4](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.3...www@1.6.4) (2020-06-20)

**Note:** Version bump only for package www





## [1.6.3](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.2...www@1.6.3) (2020-06-17)

**Note:** Version bump only for package www





## [1.6.2](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.1...www@1.6.2) (2020-06-15)

**Note:** Version bump only for package www





## [1.6.1](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.6.0...www@1.6.1) (2020-06-12)

**Note:** Version bump only for package www





# [1.6.0](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.9...www@1.6.0) (2020-05-27)


### Features

* mdx media blocks ([770618e](https://github.com/zahradnik-io/gatsby-theme-novela/commit/770618e1bef93f08ea5f951c2230ac5104e967c9))





## [1.5.9](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.8...www@1.5.9) (2020-05-27)


### Bug Fixes

* **MDXRenderer:** adding full-width for .gif ([2127daa](https://github.com/zahradnik-io/gatsby-theme-novela/commit/2127daa0f6b89f62b992181dfda64d3a8550794b))





## [1.5.8](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.7...www@1.5.8) (2020-05-22)

**Note:** Version bump only for package www





## [1.5.7](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.6...www@1.5.7) (2020-05-18)

**Note:** Version bump only for package www





## [1.5.6](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.5...www@1.5.6) (2020-05-12)

**Note:** Version bump only for package www





## [1.5.5](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.4...www@1.5.5) (2020-05-09)

**Note:** Version bump only for package www





## [1.5.4](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.3...www@1.5.4) (2020-05-08)

**Note:** Version bump only for package www





## [1.5.3](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.2...www@1.5.3) (2020-05-07)

**Note:** Version bump only for package www





## [1.5.2](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.1...www@1.5.2) (2020-05-07)

**Note:** Version bump only for package www





## [1.5.1](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.5.0...www@1.5.1) (2020-05-06)

**Note:** Version bump only for package www





# [1.5.0](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.6...www@1.5.0) (2020-05-05)


### Features

* **blog:** add Notion in social links ([da82a09](https://github.com/zahradnik-io/gatsby-theme-novela/commit/da82a0919ab23f7cdf8794febb10d9be491b7210))





## [1.4.6](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.5...www@1.4.6) (2020-05-03)

**Note:** Version bump only for package www





## [1.4.5](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.4...www@1.4.5) (2020-05-02)

**Note:** Version bump only for package www





## [1.4.4](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.3...www@1.4.4) (2020-04-30)

**Note:** Version bump only for package www





## [1.4.3](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.2...www@1.4.3) (2020-04-25)

**Note:** Version bump only for package www





## [1.4.2](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.1...www@1.4.2) (2020-04-20)

**Note:** Version bump only for package www





## [1.4.1](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.4.0...www@1.4.1) (2020-04-16)

**Note:** Version bump only for package www





# [1.4.0](https://github.com/zahradnik-io/gatsby-theme-novela/compare/www@1.3.7...www@1.4.0) (2020-04-12)


### Bug Fixes

* Fix build by changing source to local for www ([9de7d0f](https://github.com/zahradnik-io/gatsby-theme-novela/commit/9de7d0feb7aedb6105a308b12141b76aef4cf577))
* **contentful:** source works now without local ([3299592](https://github.com/zahradnik-io/gatsby-theme-novela/commit/329959220705c095ba6ce7798a3779e022bb56d6))


### Features


www/gatsby-config.js
require("dotenv").config();

const siteMetadata = {
  title: `Novela by zahradnik-io`,
  name: `Vladimír Záhradník`,
  siteUrl: `https://www.zahradnik.io/`,
  description: `This is my description that will be used in the meta tags and important for search results`,
  hero: {
    heading: `Welcome to Novela, the simplest way to start publishing with Gatsby.`,
    maxWidth: 652,
  },
  social: [
    {
      url: `https://twitter.com/VladoZahradnik`,
    },
    {
      url: `https://github.com/zahradnik-io`,
    },
    {
      url: `https://www.instagram.com/vladimir.zahradnik`,
    },
    {
      url: `https://www.linkedin.com/in/vzahradnik/`,
    },
    {
      url: `https://www.youtube.com/channel/UCogZ6qxqKa_WIsw7NnU2IaA`,
    },
    {
      name: 'stackoverflow',
      url: `https://stackoverflow.com/users/3296749/vzahradnik`,
    },
    {
      name: `mailto`,
      url: `mailto:`,
    },
  ],
};

const plugins = [
  {
    resolve: "@zahradnik-io/gatsby-theme-novela",
    options: {
      contentPosts: "content/posts",
      contentAuthors: "content/authors",
      rootPath: "/",
      basePath: "/",
      authorsPage: true,
      mailchimp: true,
      sources: {
        local: true,
        contentful: false,
      },
    },
  },
  {
    resolve: `gatsby-plugin-manifest`,
    options: {
      name: `Novela by zahradnik-io`,
      short_name: `Novela`,
      start_url: `/`,
      background_color: `#fff`,
      theme_color: `#fff`,
      display: `standalone`,
      icon: `src/assets/favicon.png`,
    },
  },
  {
    resolve: `gatsby-plugin-google-analytics`,
    options: {
      trackingId: "UA-118232427-3",
    },
  },
  {
    resolve: "gatsby-plugin-mailchimp",
    options: {
      endpoint:
        "https://narative.us19.list-manage.com/subscribe/post?u=65ef169332a03669b9538f6ef&amp;id=c55c426282",
    },
  },
];

/**
 * For development purposes if there's no Contentful Space ID and Access Token
 * set we don't want to add in gatsby-source-contentful because it will throw
 * an error.
 *
 * To enable Contentful you must
 * 1. Create a new Space on contentful.com
 * 2. Import the Contentful Model from @zahradnik-io/gatsby-theme-novela/conteful
 * 3. Add .env to www/ (see www/env.example)
 * 4. Enable contentful as a source in this file for @zahradnik-io/gatsby-theme-novela
 */
if (process.env.CONTENTFUL_SPACE_ID && process.env.CONTENTFUL_ACCESS_TOKEN) {
  plugins.push({
    resolve: "gatsby-source-contentful",
    options: {
      spaceId: process.env.CONTENTFUL_SPACE_ID,
      accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
    },
  });
}

module.exports = {
  siteMetadata,
  plugins,
};
